esmvaltool/diag_scripts/climate_patterns/sub_functions.py

```diff
@@ -0,0 +1,214 @@
+# (C) Crown Copyright 2022-2024, Met Office.
+"""Script containing relevant sub-functions for driving scripts.
+
+Author
+------
+Gregory Munday (Met Office, UK)
+"""
+
+import logging
+import multiprocessing as mp
+import os
+from functools import partial
+from pathlib import Path
+
+import iris
+import iris.analysis.cartography
+import iris.coord_categorisation
+import numpy as np
+import dask as da
+
+logger = logging.getLogger(Path(__file__).stem)
+
+
+def load_cube(filename):
+    """Load cube, remove any dimensions of length: 1.
+
+    Parameters
+    ----------
+    filename : path
+        path to load cube file
+
+    Returns
+    -------
+    cube : cube
+        a cube
+    """
+    logger.debug("Loading %s", filename)
+    cube = iris.load_cube(filename)
+    cube = iris.util.squeeze(cube)
+
+    return cube
+
+
+def ocean_fraction_calc(sftlf):
+    """Calculate gridded land and ocean fractions.
+
+    Parameters
+    ----------
+    sftlf: cube
+        land-fraction cube from piControl experiment
+
+    Returns
+    -------
+    ocean_frac: cube
+        ocean_fraction cube for area-weights
+    land_frac: cube
+        land_fraction cube for area-weights
+    of: float
+        ocean fraction float for EBM calculations
```

Comment:
Should this be removed? (Only the first two items are returned)

Reply:
Good catch, yes and removed!